databases/relational-databases.md

```diff
@@ -39,69 +77,132 @@ To understand, query, and insert information into a relational database, a techn
+    - selecting specific columns: \*, specific columns, AS column aliasing. Below are the syntax for the same.
+        ```sql
+        SELECT * FROM table_name;
+        SELECT column1, column2, ...
+        FROM table_name;
+        ```
     - modifier constraints:
       - `WHERE`
+        ```sql
+         SELECT column1, column2, ...    
+         FROM table_name  
+         WHERE condition;
+         ```
       - `ORDER BY`
+        ```sql
+          SELECT column1, column2, ...
+          FROM table_name
+          ORDER BY column1, column2, ... ASC|DESC;
+          ```
       - `LIMIT`
+        ```sql
+         SELECT * FROM table_name
+            LIMIT 3;
+         ```
       - `JOIN`
-        - Should only cover `LEFT INNER/NATURAL/JOINS` with resources pointing to other types
+        - A JOIN clause is used to combine rows from two or more tables, based on a related column between them
+        - Here are the different types of the JOINs in SQL:
+           - (INNER) JOIN: Returns records that have matching values in both tables
+           - LEFT (OUTER) JOIN: Return all records from the left table, and the matched records from the right table
+           - RIGHT (OUTER) JOIN: Return all records from the right table, and the matched records from the left table
+           - FULL (OUTER) JOIN: Return all records when there is a match in either left or right table
   - `INSERT INTO`
     - without modifiers (entire table)
+      ```sql
+       INSERT INTO table_name
+        VALUES (value1, value2, value3, ...);
+          ```
     - with modifier constraints (same as `SELECT`)
+      ```sql
+       INSERT INTO table_name (column1, column2, column3, ...)
+        VALUES (value1, value2, value3, ...);
+        ```
     - from another table `(INSERT INTO a SELECT x, y, z from b)`
     - from another table with constraints `(INSERT INTO a SELECT x, y, z FROM b WHERE x = 'a thing')`
     - Would be good to demonstrate type mismatch (`INSERT` '1' into an integer column), but if using MySQL
       to teach this doesn't work as MySQL automatically type casts depending on server config
   - `UPDATE...SET`
     - without modifiers (entire table)
+      ```sql
+       UPDATE table_name
+        SET column1 = value1, column2 = value2, ...;
+        ```
     - with modifier constraints (same as `SELECT`)
+      ```sql
+       UPDATE table_name
+        SET column1 = value1, column2 = value2, ...
+         WHERE condition;
+        ```
+
   - `DELETE FROM`
     - without modifiers (entire table)
+      ```sql
+      DELETE FROM table_name;
+      ```
     - with modifier constraints (same as `SELECT`)
+      ```sql
+      DELETE FROM table_name WHERE condition;
+      ```
+- DDL or Data Definition Language actually consists of the SQL commands that can be used to define the database schema. It simply deals with descriptions of the database schema and is used to create and modify the structure of database objects in database.
```

Comment:
https://www.geeksforgeeks.org/sql-ddl-dml-dcl-tcl-commands/
You got this from here

Reply:
👆